### PR TITLE
feat(sbtc): enroll contract call

### PR DESCRIPTION
**Pull request by kyranjamie**
Running into same issue as we've seen with Ledger, though the tx does actually broadcast

Edit; this is fixed, though I still see NaN values on the final review screen. A slight improvement at least, PR coming extension side. 

image

### CODE REVIEW

**Comment by kyranjamie** [comment]
UI when the user is enrolled already:

image

File: src/hooks/use-enroll-transaction.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+export function useSbtcEnroll() {
+  const network = useStacksNetwork();
+  const { address } = useAuth();
+
+  return useMemo(
+    () => ({
+      get enrollmentStatus() {
+        // TODO: investigate how to find enrollment state, read contract?
```
Comment (camerow) [inline src/hooks/use-enroll-transaction.ts:73]: Isn't that `useEnrolledStatus` ?

File: src/pages/choose-stacking-method/components/enroll-for-sbtc-rewards.tsx [hunk 2]
```diff
@@ -7,25 +7,34 @@ import { BridgingStepCard } from 'src/pages/choose-stacking-method/components/br
...
+    return 'Sign in to enroll';
+  }, [enrolledQuery?.isEnrolled, isSignedIn]);
 
   return (
     <>
       <BridgingStepCard
-        {...props}
         step={2}
-        disabled={!isSignedIn}
+        disabled={!isSignedIn || enrolledQuery?.isEnrolled}
```
Comment (camerow) [inline src/pages/choose-stacking-method/components/enroll-for-sbtc-rewards.tsx:31]: If they are enrolled should we link out to [bitcoinismore](http://bitcoinismore.org/) so they can see status etc?

**Comment by markmhendrickson** [comment]
@kyranjamie is this close to release for today?

**Comment by kyranjamie** [comment]
> @kyranjamie is this close to release for today? 

I need to investigate the address bug screetshot in pr 
body

**Comment by camerow** [comment]
I can run and try my best to debug

**Review by camerow — APPROVED** [review]

**Review by fbwoolf — APPROVED** [review]

**Pull request merged**